OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -2246,9 +2246,11 @@ algorithm
       Boolean b;
 
     // array equation that may result from -d=-nfScalarize and is assumed solved
-    case BackendDAE.ARRAY_EQUATION(left = DAE.CREF(componentRef = cr), right = e2, source = source, attr = eqAttr)
+    case BackendDAE.ARRAY_EQUATION(left = DAE.CREF(componentRef = cr), right = e2, source = source, attr = eqAttr) algorithm
+        varexp := Expression.crefExp(cr);
+        simEqSys := SimCode.SES_ARRAY_CALL_ASSIGN(iuniqueEqIndex, varexp, e2, source, eqAttr);
```

Comment:
seems reasonable, is this where you get the c-code error in the end because it is an array assignment?

Reply:
Yes it was creating `SimCode.SES_SIMPLE_ASSIGN` from `BackendDAE.ARRAY_EQUATION` before for some reason. I am guessing it was just a bug.  

The second commit has already fixed that. I just failed to mention it here. I think I just talked about it in the dev meeting (maybe it was just in my head :) )